src/server/replica.cc

```diff
@@ -622,10 +623,10 @@ error_code Replica::ConsumeDflyStream() {
 }
 
 void Replica::JoinAllFlows() {
-  last_journal_LSNs_.clear();
+  last_journal_LSNs_.emplace();
   for (auto& flow : shard_flows_) {
     flow->JoinFlow();
-    last_journal_LSNs_.push_back(flow->JournalExecutedCount());
+    last_journal_LSNs_->push_back(flow->JournalExecutedCount());
```

Comment:
It could be that JournalExecutedCount will return 0 right? If for some flows the full sync didnt finish

Reply:
You're right. I moved it around to the end of `ConsumeDflyStream` where it's more appropriate.